sponsorship_switzerland/models/account_invoice.py

```diff
@@ -93,3 +93,15 @@ def group_or_split_reconcile(self):
         """Reconcile given invoices with partner open payments.
         """
         return self._group_or_split_reconcile()
+
+    def action_invoice_open(self):
+        condition = [
+            ("transaction_id", "=", self.transaction_id),
+            ("state", "=", "open"),
+        ]
+        same_transactions_invoices = self.env["account.invoice"].search(condition)
+        others_same_transactions_invoices = same_transactions_invoices.filtered(lambda x: x.id != self.id)
```

Comment:
Instead of filtering your results, you can add the condition directly on your search:
```suggestion
        condition = [
            ("transaction_id", "=", invoice.transaction_id),
            ("state", "=", "open"),
            ("id", "!=", invoice.id),
        ]
```